packages/react-positioning/src/usePopper.ts

```diff
@@ -283,14 +283,16 @@ export function usePopper(
     popperInstanceRef.current?.destroy();
     popperInstanceRef.current = null;
 
+    const target = options.target || targetRef.current;
```

Comment:
We should also handle updates of `target` properly:

```diff
  useIsomorphicLayoutEffect(() => {
    handlePopperUpdate();

    return () => {
      popperInstanceRef.current?.destroy();
      popperInstanceRef.current = null;
    };
- }, [options.enabled]);
+ }, [options.enabled, options.target]);
```

Reply:
Oh that's a good catch, thanks

Reply:
I've also added the dependency to the effect which resolves popper options, WDYT ?

Reply:
> I've also added the dependency to the effect which resolves popper options, WDYT ?

It's redundant as `handleUpdate()` creates a new instance and resolves options immediately. These affects are mutually exclusive: first triggers instance recreation if it is required, another updates options if we don't need a new instance.

Reply:
reverted